### PR TITLE
Fix for memory leak in glTF2 Importer if an exception has been thrown

### PR DESCRIPTION
**Pull request by muxanickms**
Raw pointer wasn't correctly deleted if Read method throws an exception.
Unique ptr constructor is used here instead of `make_unique` because of C++11 support.

### CODE REVIEW

File: code/glTF2/glTF2Asset.inl [hunk 1]
```diff
@@ -270,13 +270,14 @@ Ref<T> LazyDict<T>::Retrieve(unsigned int i)
         throw DeadlyImportError("GLTF: Object at index \"" + to_string(i) + "\" is not a JSON object");
     }
 
-    T* inst = new T();
+    // Unique ptr prevents memory leak in case of Read throws an exception
+    auto inst = std::unique_ptr<T>();
```
Comment (turol) [inline code/glTF2/glTF2Asset.inl]: Don't you need to actually put something in `unique_ptr`? It should default-construct as empty (effectively nullptr).

Comment (muxanickms) [reply code/glTF2/glTF2Asset.inl]: you're right, thank you!

**Comment by kimkulling** [comment]
Looks good, will merge it after CI is ready!

**Comment by kimkulling** [comment]
Merged, thanks a lot for the contribution. 

I will test if some glTF2.0 issue, which had already issues with memory-leaks, are still there.

**Pull request merged**